developer-docs-site/static/messageboard_demo/client/index.ts

```diff
@@ -0,0 +1,172 @@
+
+        // publish dependency first
+        var libs = new Set<string>();
+        libs.add("ACL.mv");
+        libs.add("Offer.mv");
+        const modulePayload = this.getCompiledModuleTxnPayload(modulePath, libs);
+        await this.executeTransaction(account, modulePayload);
+
+        // publish the messageboard modules
+        var boardModules = new Set<string>();
+        boardModules.add("ACLBasedMB.mv");
+        boardModules.add("CapBasedMB.mv");
+        const boardModulePayload = this.getCompiledModuleTxnPayload(modulePath, boardModules);
+        await this.executeTransaction(account, boardModulePayload);
+
+        // create the messageboard
+        var fname = `${account.address()}::${this.boardType}::message_board_init`;
+        var args = [];
+        const initPayload = this.getScriptFunctionTxnPayload(fname, args);
+        await this.executeTransaction(account, initPayload);
+    }
+
+    async sendMessage(account: AptosAccount, message: string) {
+        var hexstring = Buffer.from(message).toString('hex');
+        console.log(`${account.address().toString()} sends message ${message} in the format of hex ${hexstring}`);
+
+        var args = [this.adminAddr.toString(), hexstring];
+        var fname = `${this.adminAddr}::${this.boardType}::send_message_to`;
+        await this.executeTransaction(
+            account,
+            this.getScriptFunctionTxnPayload(fname, args)
+        );
+    }
+
+    async getBoardEvents(): Promise<Types.Event[]> {
+        console.log("getting events from messageboard");
+        var eventHandle = `${this.adminAddr}::${this.boardType}::MessageChangeEventHandle`;
+        return await this.client.getEventsByEventHandle(
+            this.adminAddr, eventHandle, 'change_events');
+    }
+
+    async addParticipantToMessageboard(admin: AptosAccount, participant_addr: string) {
+        console.log("add participants to messageboard");
+        let args = [participant_addr];
+        let fname = `${admin.address()}::${this.boardType}::add_participant`;
+        const addParticipantPayload = this.getScriptFunctionTxnPayload(fname, args);
+        await this.executeTransaction(admin, addParticipantPayload);
+    }
+}
+
+(async () => {
+    const faucetClient = new FaucetClient(NODE_URL, FAUCET_URL, null);
+    var fakeKey = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+    var admin = new AptosAccount(new Uint8Array(fakeKey));
+    console.log(admin.address().hex());
+    var participant = new AptosAccount();
+    await faucetClient.fundAccount(admin.address(), 5000);
+    await faucetClient.fundAccount(participant.address(), 5000);
+
+    var board = new Messageboard(BoardType.ACL, admin);
+
+    // compile the modules with the admin's address
+    chdir('../../../../aptos-move/move-examples');
+    exec(
+        `cargo run -- messageboard/sources/ -a MessageBoard -h ${admin.address().toString()}`,
```

Comment:
this is really cool, we should update the documentation for first move module asap.